README.md

```diff
@@ -1,16 +1,121 @@
 
+## Core module
+
+### Dependency
+
+#### Gradle
+```groovy
+compile 'com.avast.grpc:grpc-json-bridge-akkahttp_2.12:x.x.x'
+```
+
+#### Gradle
+```scala
+libraryDependencies += "com.avast.grpc" %% "grpc-json-bridge-core" % "x.x.x"
+```
+
+### Usage
+
+Having a proto like
+```proto
+option java_package = "com.avast.grpc.jsonbridge.test";
+
+message TestApi {
+    message GetRequest {
+        repeated string names = 1;           // REQUIRED
+    }
+    
+    message GetResponse {
+        map<string, int32> results = 1;      // REQUIRED
+    }
+}
+
+service TestApiService {
+    rpc Get (TestApi.GetRequest) returns (TestApi.GetResponse) {}
+}
+```
+you can create [`GrpcJsonBridge`](core/src/main/scala/com/avast/grpc/jsonbridge/GrpcJsonBridge.scala) instance by
+```scala
+import com.avast.grpc.jsonbridge._ // this does the magic!
+
+import scala.concurrent.ExecutionContextExecutorService
+import com.avast.grpc.jsonbridge.test.TestApi
+import com.avast.grpc.jsonbridge.test.TestApi.{GetRequest, GetResponse}
+import com.avast.grpc.jsonbridge.test.TestApiServiceGrpc.{TestApiServiceFutureStub, TestApiServiceImplBase}
+import io.grpc.stub.StreamObserver
+
+implicit val executor: ExecutionContextExecutorService = ???
+
+val bridge = new TestApiServiceImplBase {
+  override def get(request: GetRequest, responseObserver: StreamObserver[TestApi.GetResponse]): Unit = {
+    responseObserver.onNext(GetResponse.newBuilder().putResults("name", 42).build())
+    responseObserver.onCompleted()
+  }
+}.createGrpcJsonBridge[TestApiServiceFutureStub]() // this does the magic!
+```
+or you can even go with the [Cactus](https://github.com/avast/cactus) and let it map the GPB messages to your case classes:
+```scala
+import com.avast.grpc.jsonbridge._ // import for the grpc-json-bridge mapping
+import com.avast.cactus.grpc.server._ // import for the cactus mapping
+
+import com.avast.grpc.jsonbridge.test.TestApiServiceGrpc.{TestApiServiceFutureStub, TestApiServiceImplBase}
+import io.grpc.Status
+
+import scala.concurrent.{ExecutionContextExecutorService, Future}
+
+implicit val executor: ExecutionContextExecutorService = ???
+
+case class MyRequest(names: Seq[String])
+
+case class MyResponse(results: Map[String, Int])
+
+trait MyApi {
+  def get(request: MyRequest): Future[Either[Status, MyResponse]]
+}
+
+val service = new MyApi {
+  override def get(request: MyRequest): Future[Either[Status, MyResponse]] = Future.successful {
+    Right {
+      MyResponse {
+        Map(
+          "name" -> 42
+        )
+      }
+    }
+  }
+}.mappedToService[TestApiServiceImplBase]() // cactus mapping
+
+val bridge = service.createGrpcJsonBridge[TestApiServiceFutureStub]()
+```
+
+### Calling the bridged service
+
+You can use e.g. cURL command to call the `Get` method
+
+```
+curl -H "Content-Type: application/json" --data " { \"names\": [\"abc\",\"def\"] } " http://localhost:9999/com.avast.grpc.jsonbridge.test.TestApiServiceGrpc.TestApiServiceImplBase/Get
```

Comment:
Add `-X POST` here to stress that a `POST` method is expected. As the gRPC operation is named `Get`, one might easily confuse it with a `GET` method.

Reply:
I removed that because the cURL shows a warning it's redundant since the POST method is inferred.

Reply:
Done